modules/EmailMan/EmailMan.php

```diff
@@ -609,6 +622,14 @@ public function sendEmail($mail, $save_emails = 1, $testmode = false)
         $module->retrieve($this->related_id);
         $module->emailAddress->handleLegacyRetrieve($module);
 
+        if ($this->shouldBlockEmail($module)) {
+            $GLOBALS['log']->debug('Email Address was sent due to not being confirm opt in' . $module->email1);
+
+            // block sending campaign email
+            $this->set_as_sent($module->email1, true, null, null, 'blocked');
```

Comment:
This section follows the same convention has the rest of the method. See line 637 for example. sendEmail is assumes that the module is a contact, account, user, lead, or target. All of these modules have an email1. 

The Emails modules utilises a different set of methods to send an email. 

So this is valid, because SuiteCRM is supposed to send an email to the primary email address of a record.

However, I guess I should have put it below the checks for the primary email address and the validation. So i have moved it.

Reply:
Have a look at line 618
```php
//check to see if bean has a primary email address
if (!$this->is_primary_email_address($module)) {
```

is_primary_email_address provides the guarantee. Bear in mind, I have moved my check after this. The shouldBlockEmail method is assuming that the email1 exists in the module, because the is_primary_email_address has already validated it.

Reply:
If I add a bean to these function which has not `email1` field, these function shows a php notice. Probably also need an `isset($foo->email1)` also.

Reply:
I will add the isset to the is_primary_email_address and that will remove the notice and ensure that the email1 exists.